game/src/ecs/components/ai/transition/ProtectOnApproach.java

```diff
@@ -0,0 +1,46 @@
+package ecs.components.ai.transition;
+
+import ecs.components.ai.AITools;
+import ecs.entities.Entity;
+
+/**
+ * Implements an AI that protects one or more specific entities if hero is in the given range
+ *
+ * <p>Entity will stay in fight mode once it entered it</p>
+ */
+public class ProtectOnApproach implements ITransition {
+    private final float range;
+
+    private final Entity toProtect;
+
+    private boolean isInFight = false;
+
+    /**
+     * Constructor needs a range and the entity to protect.
+     *
+     * @param range     - The range in which the entity should in fight mode
+     * @param toProtect - The entity which should be protected
+     */
+    public ProtectOnApproach(float range, Entity toProtect) {
```

Comment:
In der Klassen-Javadoc wird von "mehrere" Entitäten gesprochen, implementiert ist es aber nur für eine